MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs

```diff
@@ -425,7 +425,13 @@ private async Task ConvertTextSubtitleToSrtInternal(string inputPath, string lan
 
             var encodingParam = await GetSubtitleFileCharacterSet(inputPath, language, inputProtocol, cancellationToken).ConfigureAwait(false);
 
-            if (!string.IsNullOrEmpty(encodingParam))
+            // FFmpeg automatically convert character encoding when it is UTF-16
+            // If we specify character encoding, it rejects with "do not specify a character encoding" and "Unable to recode subtitle event"
+            if ((inputPath.EndsWith(".smi") || inputPath.EndsWith(".sami")) && (encodingParam == "UTF-16BE" || encodingParam == "UTF-16LE"))
```

Comment:
Might be worth it to link to the issue #1539 in this comment, so when someone later sees this hack they know not to touch it. Not sure how @jellyfin/core feels about that?

Reply:
Would it better to add a new commit or force-push after amending, to make issue number in comments?

Reply:
I think it would be fine without tagging the issue. Does this change make sense for any subtitle format or only those two?

Reply:
Actually, the srt subtitles are affected when I see the code of ffmpeg and tested with ffmpeg. 
But srt subtitles aren't converted to srt (as it is already srt), so in jellyfin, it works.  
So I didn't added srt to the extensions list.